File: Leetcode/242 - Valid Anagram/solution.py
def isAnagram(self, s: str, t: str) -> bool:
    if len(s) != len(t):
        return False

    char_count: Dict[str, int] = {}

    for i in range(len(s)):
        char_count[s[i]] = char_count.get(s[i], 0) + 1
        char_count[t[i]] = char_count.get(t[i], 0) - 1

    for cnt in char_count.values():
        if cnt != 0:
            return False

    return True

File: Leetcode/242 - Valid Anagram/solution.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) return false;

    const charCount: Record<string, number> = {};
    for (let i = 0; i < s.length; i++) {
        charCount[s[i]] = (charCount[s[i]] ?? 0) + 1;
        charCount[t[i]] = (charCount[t[i]] ?? 0) - 1;
    }

    for (const count of Object.values(charCount)) {
        if (count != 0) {
            return false;
        }
    }

    return true;
};